ratis-client/src/main/java/org/apache/ratis/client/impl/SnapshotManagementImpl.java

```diff
@@ -27,15 +28,19 @@
 
 class SnapshotManagementImpl implements SnapshotManagementApi {
   private final RaftClientImpl client;
+  private final RaftPeerId server;
 
-  SnapshotManagementImpl(RaftClientImpl client) {
+  SnapshotManagementImpl(RaftPeerId server, RaftClientImpl client) {
+    this.server = Objects.requireNonNull(server, "server == null");
     this.client = Objects.requireNonNull(client, "client == null");
   }
 
   @Override
   public RaftClientReply create(long timeoutMs) throws IOException {
     final long callId = CallId.getAndIncrement();
-    return client.io().sendRequestWithRetry(() -> SnapshotManagementRequest.newCreate(
-        client.getId(), client.getLeaderId(), client.getGroupId(), callId, timeoutMs));
+    final RaftClientReply reply = client.io().sendRequestWithRetry(
+        () -> SnapshotManagementRequest.newCreate(client.getId(), server,
+            client.getGroupId(), callId, timeoutMs));
+    return reply;
```

Comment:
We should support the case the server == null. When server == null, use client.getLeaderId() as before.
```
    return client.io().sendRequestWithRetry(() -> SnapshotManagementRequest.newCreate(client.getId(),
        Optional.ofNullable(server).orElseGet(client::getLeaderId), client.getGroupId(), callId, timeoutMs));
```